### PR TITLE
Make inference_api_test compile with shared lib in Windows

### PR DESCRIPTION
**Pull request by betterpig**
### PR types

Others
### PR changes

Others
### Describe

Make inference_api_test compile with shared lib in windows. The original PR #41944, it is reverted because ROCM and kunlun will fail in cmake after PR merged.
image

Note: 
When CI_SKIP_CPP_TEST=ON, will not build and test cpp test. So in [paddle/fluid/inference/tests/api/CMakeLists.txt](https://github.com/PaddlePaddle/Paddle/blob/692466b4c700b9e369efead356aebf8a2022f9fb/paddle/fluid/inference/tests/api/CMakeLists.txt#L1227), cmake should return right now after finding CI_SKIP_CPP_TEST=ON. And that's the reason why ROCM and kunlu failed in cmake after PR #41944 merged.

### CODE REVIEW

**Review by tianshuo78520a — APPROVED** [review]
LGTM

**Review by jiweibo — APPROVED** [review]
LGTM

File: paddle/fluid/inference/tests/api/CMakeLists.txt [hunk 1]
```diff
@@ -1,13 +1,9 @@
...
-                           ir_pass_manager analysis_predictor benchmark)
+# If CI_SKIP_CPP_TEST=ON, there is no need to build and run these test.
+if("$ENV{CI_SKIP_CPP_TEST}" STREQUAL "ON")
+  return()
 endif()
 
-if(WITH_GPU AND TENSORRT_FOUND)
-  set(INFERENCE_EXTRA_DEPS ${INFERENCE_EXTRA_DEPS} analysis ${analysis_deps})
-endif()
+set(INFERENCE_EXTRA_DEPS paddle_inference_shared)
```
Comment (zhwesky2010) [inline paddle/fluid/inference/tests/api/CMakeLists.txt:6]: 这个变量去掉了

**Review by zhwesky2010 — APPROVED** [review]

File: paddle/fluid/inference/tests/api/CMakeLists.txt (hunk 1 above)
Comment (betterpig) [reply paddle/fluid/inference/tests/api/CMakeLists.txt:6]: 好的，后面会去掉。

**Pull request merged**